vllm/v1/executor/multiproc_executor.py

```diff
@@ -172,16 +172,23 @@ def wait_for_termination(procs, timeout):
 
         # Send SIGTERM if still running
         active_procs = [w.proc for w in self.workers if w.proc.is_alive()]
-        self.workers = None
         for p in active_procs:
             p.terminate()
-        if wait_for_termination(active_procs, 4):
-            return
+        if not wait_for_termination(active_procs, 4):
+            # Send SIGKILL if still running
+            active_procs = [p for p in active_procs if p.is_alive()]
+            for p in active_procs:
+                p.kill()
 
-        # Send SIGKILL if still running
-        active_procs = [p for p in active_procs if p.is_alive()]
-        for p in active_procs:
-            p.kill()
+        self._cleanup_sockets()
```

Comment:
Maybe we clean up the sockets before we terminate the workers, in case the client process sends a SIGKILL to this one while this process waits for the workers to gracefully terminate?

Reply:
Cleaning up before terminating workers is more likely to fail because the workers may still have the socket open.

I think anywhere we use SIGKILL is giving up the illusion of proper cleanup. Hopefully that's rare?